### DISCUSSION TITLE
Suggestion: Add Archiving Note for archived repositories

### DISCUSSION THREAD

**Discussion by Marcono1234** [community/community#5383] [Other Feature Feedback, Questions, & Ideas]
## Problem this suggestion tries to solve
Often when GitHub repositories are archived, it is not obvious ...
- why the repository has been archived
- whether there are alternative or successor projects

The "About" section of a repository is not (on its own) well suited for this information because:
- It is rather small and does not support any advanced formatting
- It is displayed in the top right corner of a repository and easy to overlook

Users often edit the README of the project to indicate why it has been archived, but that has the following issues:
- There is no standard way in the README to indicate why a repository has been archived; every repository does it differently (and some don't do it at all)
- Some repositories delete the complete repository content in the HEAD commit and only add a README containing the archiving reason; this makes it annoying for users who want to check out or clone the archived repository (it might also break GitHub links which are not permalinks)
- The README cannot be edited anymore after the repository has been archived. Any changes (e.g. fixing a link to the successor project) require to un-archive the repository first.

## Suggestion
There should be a standard way for repositories to describe why they have been archived (an "Archiving Note").
It should behave this way:
- When users archive a repository they are prompted for an Archiving Note. The 'hint' text of the text field should suggest what the user can write (e.g. "Why has the repository been archived? Is there a successor project? Are there alternatives?"). Users may leave the Archiving Note empty.
- The Archiving Note should support (basic) Markdown.
- The Archiving Note should appear prominently on the repository.
  - Possibly also on the "Releases" page, as [suggested in the comments](https://github.com/orgs/community/discussions/5383#discussioncomment-15397336).
- The Archiving Note should be editable at any point, not requiring to un-archive the repository first.
- Archived repositories without Archiving Note should be able to add one retroactively.

Here is a mockup of how such an archiving note could look:
![Archiving Note mockup](https://user-images.githubusercontent.com/11685886/131539079-0d157583-3fde-410b-9edc-2603af37cf54.png)

Maybe users viewing the archived repository should also have the option to collapse the note (to some extent) so it does not hinder them too much when they want to browse the source of the archived repository.

**Comment by Ashark** [comment]
I think this makes sense.

One another reason one may not want to edit README (for adding an archiving note) is that you may want to preserve the history of project untouched.

**Comment by Jarvis-Ank** [comment]
This should be one of the features, still following the same old way of updating readme.

**Comment by SirMangler** [comment]
Another angle to this problem, developers who archive their repositories then become unable to add any warning to users that the project has become archived. 

I have a project that was archived over 3 years ago, and even till this day I constantly have users making issue reports and joining support channels for related projects to report that the project is not functional. Very few users take notice of the existing archive notice, and those that do, do not understand the significance of it being archived to begin with.

Currently, the only solution we have is to **unarchive** the project, and then modify the release description to add a large banner dissuading users and to redirect them. 

It's very important to bare in mind, as individuals with a ton of experience on GitHub, we intuitively check the README, and many developers make the mistake of forgetting that _most users don't even realize home/readme page exists to begin with_, because most traffic goes to the release page. This means any change to the README, and any change to the 'About' section is largely pointless for a lot of users. I would like to see this archiving note, and I believe it should also exist above the release list as well for aforementioned reasons.
